pylint/lint/pylinter.py

```diff
@@ -690,18 +693,18 @@ def set_option(self, optname, value, action=None, optdict=None):
                     meth(value)
                 return  # no need to call set_option, disable/enable methods do it
         elif optname == "output-format":
-            self._reporter_names = value
-            # If the reporters are already available, load
-            # the reporter class.
-            if self._reporters:
-                self._load_reporters()
-
+            if not isinstance(value, str):
+                raise optparse.OptionValueError(
```

Comment:
This line is not tested.

Reply:
I'm not sure how to handle this:
Whenever we encounter this option it should always be `str`. Only if you manipulate it by setting the option directly and supplying a `list` (for example) this would raise the warning. In normal use cases that shouldn't be possible. However, `mypy` will complain if we don't assert that it is a `str`...
I could add a test but I foresee that we will then need to start adding tests for every option + incorrect value type. That also seems counter-productive..

Reply:
Should we use ``assert isinstance(value, str)`` instead then ?

Reply:
👍

Reply:
Added!